libraries/chain/asset_evaluator.cpp

```diff
@@ -549,7 +554,18 @@ void_result asset_publish_feeds_evaluator::do_apply(const asset_publish_feed_ope
    });
 
    if( !(old_feed == bad.current_feed) )
+   {
+      if( bad.has_settlement() ) // implies head_block_time > HARDFORK_CORE_216_TIME
+      {
+         const auto& mia_dyn = base.dynamic_asset_data_id(d);
+         if( !bad.current_feed.settlement_price.is_null()
+             && ~price::call_price(asset(mia_dyn.current_supply, o.asset_id),
+                                   asset(bad.settlement_fund, bad.options.short_backing_asset),
+                                   bad.current_feed.maintenance_collateral_ratio ) < bad.current_feed.settlement_price )
```

Comment:
The feed price is defined to be debt/collateral, see asset.hpp. The call price is computed as the inverse of debt/collateral*ratio. So if the inverted call price is lower than the feed, it means that we have less debt per collateral than required.